### ISSUE TITLE
Konsti_day2_boundary_loss

### ISSUE THREAD

**Issue by KonstiNik** [zincware/ZnPINNLab#13]
## Markdown task
I am missing a bit of guidance in the markdown task of the boundary loss markdown task. 
We do introduce the system in the beginning, but I don't think it is too clear why we now need boundary conditions.

I think we need to tell that 
- we want to solve the time-dependent SE
- For that we need to tell the solution of the time evolution about symmetry
- this is why add boundary points that constrain the symmetry

## Docstring (already adapted)
Moreover, I adjusted the docstring of the `compute_boundary_loss` function:
```python
def compute_boundary_loss(model, dataset: dict):
    """
    Define the boundary loss, representing one part of the physics-informed loss for the PINN model.
    
    This function calculates the loss used to incorporate the boundary conditions
    into the PINN. The following steps are involved:
    
    Step 1: Data Preparation
        - Get the 'boundary''inputs' tensors from the dataset using appropriately named variables.

    Step 2: Model Prediction
        - Concatenate the two input tensors along axis 1.
        - Using the concatenated tensors, predict the real and imaginary values with the neural network model.

    Step 3: Gradient Computation
        - Compute the first gradients 'hp_r_x', 'hp_i_x', 'hn_r_x', and 'hn_i_x' 
          with respect to the 'x' input tensor using the 'torch.autograd.grad' method.

    Step 4: Difference Calculation
        - Calculate the difference between 'hp' (h at x=5) and 'hn' (h at x=-5) for real and imaginary parts.
        - Use 'torch.unsqueeze' to add a dimension along the 1 axis.

    Step 5: Derivative Difference Calculation
        - Calculate the difference between 'hp_x' and 'hn_x' for real and imaginary parts.
        - This time no 'torch.unsqueeze' needed

    Step 6: Residual Definition
        - Define 'h_err' and 'h_x_err' by adding the squared differences/ sum previously calculated.

    Step 7: Loss Calculation
        - Compute the mean squared error loss for 'h_err' and 'h_x_err' using the 'nn.MSELoss' class.
        - as a target serves a tensor containing zeros. It can be produced using 'torch.zeros_like'
        - Return the combined physics-informed loss as the sum of 'loss_h' and 'loss_h_x'.
"""
```
